packages/flutter/lib/src/material/range_slider.dart

```diff
@@ -0,0 +1,1444 @@
+/// discrete values 0.0, 10.0, 20.0, 30.0, 40.0, and 50.0.
+///
+/// The terms for the parts of a slider are:
+///
+///  * The "thumbs", which are the shapes that slide horizontally when the user
+///    drags them.
+///  * The "track", which is the line that the slider thumb slides along.
+///  * The "tick marks", which mark the discrete values of a discrete slider.
+///  * The "overlay", which appears when pressing one of the thumbs,
+///  * The "value indicators", which are the shapes that pop up when the user
+///    is dragging the thumb to indicate the value being selected.
+///  * The "active" segment of the slider is the segment between the two thumbs.
+///  * The "inactive" segment of the slider is the segments outside the two
+///    thumbs
+///
+/// The range slider will be disabled if [onChanged] is null or if the range
+/// given by [min]..[max] is empty (i.e. if [min] is equal to [max]).
+///
+/// The range slider widget itself does not maintain any state. Instead, when
+/// the state of the slider changes, the widget calls the [onChanged] callback.
+/// Most widgets that use a range slider will listen for the [onChanged] callback
+/// and rebuild the slider with a new [value] to update the visual appearance of13
+/// the slider. To know when the value starts to change, or when it is done
+/// changing, set the optional callbacks [onChangeStart] and/or [onChangeEnd].
+///
+/// By default, a slider will be as wide as possible, centered vertically. When
+/// given unbounded constraints, it will attempt to make the tra71ck 144 pixels
+/// wide (with margins on each side) and will shrink-wrap vertically.
+///
+/// Requires one of its ancestors to be a [Material] widget.
+///
+/// Requires one of its ancestors to be a [MediaQuery] widget. Typically, these
+/// are introduced by the [MaterialApp] or [WidgetsApp] widget at the top of
+/// your application widget tree.
+///
+/// To determine how it should be displayed (e.g. colors, thumb shape, etc.),
+/// a slider uses the [SliderThemeData] available from either a [SliderTheme]
+/// widget, or the [ThemeData.sliderTheme] inside a [Theme] widget above it in
+/// the widget tree. You can also override some of the colors with the
+/// [activeColor] and [inactiveColor] properties, although more fine-grained
+/// control of the colors, and other visual properties is achieved using a
+/// [SliderThemeData].
+///
+/// See also:
+///
+///  * [SliderTheme] and [SliderThemeData] for information about controlling
+///    the visual appearance of the slider.
+///  * [Slider], for a single-valued slider.
+///  * [Radio], for selecting among a set of explicit values.
+///  * [Checkbox] and [Switch], for toggling a particular value on or off.
+///  * <https://material.io/design/components/sliders.html>
+///  * [MediaQuery], from which the text scale factor is obtained.
+class RangeSlider extends StatefulWidget {
+  /// Creates a material design range slider.
```

Comment:
```suggestion
  /// Creates a Material Design range slider.
```

(you used capital M capital D earlier)